keras/src/models/model_test.py

```diff
@@ -1237,3 +1239,141 @@ def test_export_error(self):
+        def __init__(self, embed_dim, num_heads, ff_dim, **kwargs):
+            super().__init__(**kwargs)
+            # The standard MultiHeadAttention layer uses Dense layers
+            # for its projections.
+            self.att = layers.MultiHeadAttention(
+                num_heads=num_heads, key_dim=embed_dim
+            )
+            self.ffn = models.Sequential(
+                [
+                    layers.Dense(ff_dim, activation="relu"),
+                    layers.Dense(embed_dim),
+                ]
+            )
+            self.layernorm1 = layers.LayerNormalization(epsilon=1e-6)
+            self.layernorm2 = layers.LayerNormalization(epsilon=1e-6)
+
+        def call(self, inputs):
+            attention_output = self.att(inputs, inputs)
+            out1 = self.layernorm1(inputs + attention_output)
+            ffn_output = self.ffn(out1)
+            return self.layernorm2(out1 + ffn_output)
+
+    inputs = layers.Input(shape=(None,), dtype="int32")
+    embedding_layer = layers.Embedding(vocab_size, embed_dim)
+    x = embedding_layer(inputs)
+    transformer_block = SimpleTransformerBlock(embed_dim, num_heads, ff_dim)
+    x = transformer_block(x)
+    outputs = layers.Dense(vocab_size)(x)
+    model = models.Model(inputs=inputs, outputs=outputs)
+    return model
+
+
+@pytest.mark.requires_trainable_backend
+class ModelQuantizationTest(testing.TestCase):
+    def _run_gptq_test_on_dataset(self, dataset):
+        """Helper function to run a full GPTQ quantization
+        test on a given dataset."""
+
+        model = _get_model_with_dense_attention()
+
+        # 1. Common setup
+        NUM_SAMPLES = 16
+        SEQUENCE_LENGTH = 128
+        VOCAB_SIZE = 1000
+        W_BITS = 4
+        GROUP_SIZE = 32
+
+        mock_tokenizer = lambda text: np.array(
+            [ord(c) % VOCAB_SIZE for c in text]
+        )
+        mock_tokenizer.tokenize = mock_tokenizer
+
+        # 2. Find target layer and get original weights
+        target_layer = model.layers[2].ffn.layers[0]
+
+        self.assertIsNotNone(
+            target_layer,
+            "Test setup failed: No Dense layer was found inside "
+            "an 'ffn' block.",
+        )
+        original_weights = np.copy(target_layer.kernel.numpy())
+
+        # 3. Configure and run quantization
+        gptq_config = GPTQConfig(
+            dataset=dataset,
+            tokenizer=mock_tokenizer,
+            wbits=W_BITS,
+            nsamples=NUM_SAMPLES,
+            seqlen=SEQUENCE_LENGTH,
+            groupsize=GROUP_SIZE,
+        )
+        model.quantize("gptq", quant_config=gptq_config)
+
+        #  4. Assertions and verification
+        quantized_weights = target_layer.kernel.numpy()
+
+        # Assert that the weights have been changed
+        self.assertFalse(
+            np.allclose(original_weights, quantized_weights),
+            "Weights were not changed by the GPTQ process for "
+            "dataset: {dataset}",
+        )
```

Comment:
We can simply replace this with a call to `self.assertNotAllClose`
```suggestion
        self.assertNotAllClose(original_weights, quantized_weights,
            msg="Weights were not changed by the GPTQ process for dataset: {dataset}",
        )
```

Reply:
Done